### DISCUSSION TITLE
Bringing you attention to MFront, an easy-to-use and advance "Material Frontend"

### DISCUSSION THREAD

**Discussion by Irvise** [OpenRadioss/OpenRadioss#2943] [Ideas]
As the title says, I would like to bring [TFEL/MFront](https://thelfer.github.io/tfel/web/index.html) to the attention of the OpenRADIOSS developers and wider community as it may be of your interest. As a short summary, MFront is an open source tool designed to easily create material behaviours/constitutive laws in an easy way (read further for examples). The motivation to write this post comes after watching Paul Du Bois talk in last years' Aachen OpenRADIOSS meeting. I think MFront serves as a great framework that could enable the quick development of new and advance material laws (in contrast to `LUSERnn`) and it may be worth for the developers of OpenRADIOSS to check it out.

It is developed by CEA (Center of Atomic Energies) and EDF (Electrical supplier of France) and, AFAIK, it was created to model nuclear fuels (thermo-mechanically). Since then it has evolved to be a generic tool with which users could create advance material models in a much easier to use manner than, say a Fortran UMAT subroutine or `LUSERnn` law. MFront currently supports generating behaviours for several well-known programs, such as ANSYS, ABAQUS, ABAQUS/Explicit, [Code_Aster](https://code-aster.org/) and others. So one would need to write the material model once, and be able to target several different programs if one desires it. The performance of the generated models is also really good (see their own benchmarks and comparisons). It also supports a [generic interface](https://github.com/thelfer/MFrontGenericInterfaceSupport).

A couple of examples that show how powerful and capable it is, can be found in the [examples/gallery repository](https://github.com/thelfer/MFrontGallery). For example, it is fairly easy to create [chemical reaction kinematics](https://github.com/thelfer/MFrontGallery/blob/master/mfront-book-examples/models/ChemicalReaction1.mfront), [crushed salt](https://github.com/thelfer/MFrontGallery/blob/master/materials/CrushedSalt/behaviours/CrushedSaltKorthausBehaviour.mfront) or [viscoplastic damage](https://github.com/thelfer/MFrontGallery/blob/master/generic-behaviours/damage_viscoplasticity/FichantLaBorderieDamageAndTorelliLoadInducedThermalStrainBehaviour.mfront) models. But all the aforementioned examples are written from scratch, that is advance stuff. MFront brings flexible and very capable built-in behaviours that are "fill-in the blank" easy. As an example, take a look at the [Standard ElastoViscoPlastic "Brick"](https://thelfer.github.io/tfel/web/StandardElastoViscoPlasticityBrick.html) or the [Finite Strain Single Cristal model](https://thelfer.github.io/tfel/web/finitestrainsinglecrystal.html).

So as a summary, it could be a nice addition for the (Open)RADIOSS ecosystem to have such a nice an powerful tool, that would enable rapid and efficient development of constitutive material laws. And I know that RADIOSS/Altair are involved in these kinds of activities :)

If you have any questions, I hope to be able to provide answers. Otherwise, the Thomas, the main developer of MFront is a really nice and helpful person ^^

Best regards,
Fer

**Comment by MarianBulla** [comment]
Dear @Irvise, 
Thank you very much for your interest in OpenRadioss.
It sounds to be really a great tool.

I'm interested to learn more about TFEL/MFront. Would it be possible for you to have a call/Teams meeting discussing little bit more about it ?

Please, contact me at communitymanager@openradioss.org for more details.

Thanks and best regards,
Marian
